Cargo.toml

```diff
@@ -62,6 +62,7 @@ rand_xorshift = "0.4.0"
 rand_xoshiro = "0.7.0"
 serde = "1.0"
 serde_json = "1.0.142"
+num_cpus = "1.17.0"
```

Comment:
I'm already using `sysinfo = { workspace = true }` in `slatedb-dst`. Mind using that crate to fetch CPU counts?

https://docs.rs/sysinfo/latest/sysinfo/struct.System.html#method.cpus